x-pack/plugins/ingest_manager/common/services/is_agent_upgradeable.test.ts

```diff
@@ -135,4 +135,35 @@ describe('Ingest Manager - isAgentUpgradeable', () => {
       true
     );
   });
+  it('returns false if agent reports upgradeable, with agent snapshot version === kibana version', () => {
+    expect(
+      isAgentUpgradeable(getAgent({ version: '7.9.0-SNAPSHOT', upgradeable: true }), '7.9.0')
+    ).toBe(false);
+  });
+  it('returns false if agent reports upgradeable, with agent version === kibana snapshot version', () => {
+    expect(
+      isAgentUpgradeable(getAgent({ version: '7.9.0', upgradeable: true }), '7.9.0-SNAPSHOT')
+    ).toBe(false);
+  });
+  it('returns true if agent reports upgradeable, with agent snapshot version < kibana snapshot version', () => {
+    expect(
+      isAgentUpgradeable(
+        getAgent({ version: '7.9.0-SNAPSHOT', upgradeable: true }),
+        '8.0.0-SNAPSHOT'
+      )
+    ).toBe(true);
+  });
+  it('returns false if agent reports upgradeable, with agent snapshot version === kibana snapshot version', () => {
+    expect(
+      isAgentUpgradeable(
```

Comment:
i'm wondering whether this should be upgradable. 
e.g you're running snapshot 8.0.0 but want to update to todays snapshot

Reply:
in the UI we only show upgradeable if the version is less than the kibana version, regardless of whether its a snapshot.   Also the agent version should never have the '-SNAPSHOT' prefix in it (it does not currently), I just put it in case that were to change.  It currently only reports the number in the metadata.

If we want to make an exception where if an agent is a snapshot that the versions can be equal, I would need to check the local metadata to see whether its a snapshot and consider it upgradeable when versions are the same.  I am find with doing that, but not sure how common this case would be.  I'd rather add it as an enhancement in the next version.